Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,22 +25,70 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+
+     return amphoursUsed;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+if(current<0)
+{
+    return true;
+}
+else
+{return false;}
+
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+
+    double timeUntilCharged;
+    double timeUntilDepletion;
+
+   QTime charged  (0,0,0,0);
+   timeUntilCharged= (amphoursUsed / current)*(3600 * 1000);
+
+   charged = charged.addMSecs(timeUntilCharged);
+   QTime depletion (0,0,0,0);
+   timeUntilDepletion=((BATTERY_AMP_HOUR_CAPACITY-amphoursUsed)*3600000)/current;
+   depletion = depletion.addMSecs(qAbs(timeUntilDepletion));
+
+   if (isCharging()==true){
```

Comment:
Fix the indentation as mentioned before.

Also, there's a way to check this without the `==true` part. See if you can't figure out how!